coffee/base.py

```diff
@@ -221,20 +217,61 @@ class ArrayInit(Expr):
     to some values. For example ::
 
         A[3][3] = {{0.0}} or A[3] = {1, 1, 1}.
+    """
+
+    _default_precision = 12
 
-    At the moment, initial values like ``{{0.0}}`` and ``{1, 1, 1}`` are passed
-    in as simple strings."""
+    def __init__(self, values, precision=None):
+        """Initialize an ArrayInit object.
 
-    def __init__(self, values):
+        :arg values: representation of the values the array is initialized to
+        :type values: a string or a numpy ndarray.
+        :arg precision: the number of decimal digits that should be used when
+            converting a float (in a numpy array) to a string.
+        :type precision: integer (defaults to 12)
+        """
         self.values = values
+        self.precision = precision or ArrayInit._default_precision
 
-    def reconstruct(self, values, **kwargs):
-        return type(self)(values, **kwargs)
+    def reconstruct(self, values, precision, **kwargs):
+        return type(self)(values, precision, **kwargs)
 
     def operands(self):
         return [self.values], {}
 
+    @property
+    def values(self):
+        return self._values
+
+    @values.setter
+    def values(self, val):
+        if not isinstance(val, (np.ndarray, str)):
+            raise TypeError
+        self._values = val
+
+    def _formatter(self, v):
+        """Format a float into a string, showing up to ``precision`` decimal digits.
+        This function is partly extracted from the open_source "FFC: the FEniCS Form
+        Compiler", freely accessible at https://bitbucket.org/fenics-project/ffc."""
+        f = "%%.%dg" % self.precision
+        f_int  = "%%.%df" % 1
+        eps = eval("1e-%s" % self.precision)
+        return f_int % v if abs(v - round(v, 1)) < eps else f % v
```

Comment:
I want:
- values to always be in float format (1.0 etc, not 1)
- values to show up to ``self.precision`` digits after the '.' , where up to means that if you have 1.0000, you only show 1.0